lib/auth/bot.go

```diff
@@ -174,6 +178,21 @@ func (a *Server) validateGenerationLabel(ctx context.Context, username string, c
 	return nil
 }
 
+func sshPublicKeyToPKIXPEM(pubKey []byte) ([]byte, error) {
+	cryptoPubKey, err := sshutils.CryptoPublicKey(pubKey)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	pkix, err := x509.MarshalPKIXPublicKey(cryptoPubKey)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	return pem.EncodeToMemory(&pem.Block{
+		Type:  keys.PKIXPublicKeyType,
+		Bytes: pkix,
+	}), nil
```

Comment:
we often marshal RSA keys as PKCS#1 in `RSA PUBLIC KEY` PEM blocks and all other keys as PKIX in `PUBLIC KEY` PEM blocks, it might be more consistent of us to continue with that. And actually I'll be updating `certRequest` to include a TLS public key in that format very soon, so we won't even need to reformat here.

You could use `keys.MarshalPublicKey` here https://github.com/gravitational/teleport/blob/f62149ca90223f487c77db419d11f6297d0d139d/api/utils/keys/publickey.go#L37

Reply:
Yeah - part of me really likes preferring PKIX because it's uniform - but I do agree that you are probably right here and we ought to be consistent w/ the rest of Teleport. I'll switch to MarshalPublicKey.

Reply:
[Prefer using keys.MarshalPublicKey](https://github.com/gravitational/teleport/pull/43845/commits/806816d9368e60ba433f5e41f3c4f86925cb2b27)